source/Calamari.Common/Features/Packages/Decorators/ArchiveLimits/EnforceCompressionRatioDecorator.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using SharpCompress.Archives;
+
+namespace Calamari.Common.Features.Packages.Decorators.ArchiveLimits
+{
+    /// <summary>
+    /// This decorator prevents extraction of an archive if it would exceed the theoretical limits of current compression algorithms.
+    /// </summary>
+    public class EnforceCompressionRatioDecorator : PackageExtractorDecorator
+    {
+        readonly int maximumCompressionRatio;
+
+        public EnforceCompressionRatioDecorator(IPackageExtractor concreteExtractor, int maximumCompressionRatio) 
+            : base(concreteExtractor)
+        {
+            this.maximumCompressionRatio = maximumCompressionRatio;
+        }
+
+        public override int Extract(string packageFile, string directory)
+        {
+            if (!TryVerifyCompressionRatio(packageFile, out var archiveLimitException))
+                throw archiveLimitException!;
+
+            return base.Extract(packageFile, directory);
+        }
+
+        bool TryVerifyCompressionRatio(string packageFile, out ArchiveLimitException? exception)
+        {
+            try
+            {
+                if (maximumCompressionRatio >= 1)
```

Comment:
I'm assuming this is more of a safety belt to ensure that this maximum compression ratio is set (as intended)? 👍 👍

Reply:
Yep just ensuring nothing nonsense is provided.